driver/modern_bpf/helpers/store/auxmap_store_params.h

```diff
@@ -418,3 +439,314 @@ static __always_inline void auxmap__store_path_from_fd(struct auxiliary_map *aux
+		struct sockaddr_in6 *sockaddr_in6 = (struct sockaddr_in6 *)sockaddr;
+
+		/* Copy address and port into the stack. */
+		u32 ipv6[4] = {0, 0, 0, 0};
+		__builtin_memcpy(&ipv6, sockaddr_in6->sin6_addr.in6_u.u6_addr32, 16);
+		u16 port = sockaddr_in6->sin6_port;
+
+		/* Pack the sockaddr info:
+		 * - socket family.
+		 * - dest_ipv6.
+		 * - dest_port.
+		 */
+		push__u8(auxmap->data, &auxmap->payload_pos, socket_family_to_scap(socket_family));
+		push__ipv6(auxmap->data, &auxmap->payload_pos, ipv6);
+		push__u16(auxmap->data, &auxmap->payload_pos, ntohs(port));
+		final_param_len = FAMILY_SIZE + IPV6_SIZE + PORT_SIZE;
+		break;
+	}
+
+	case AF_UNIX:
+	{
+		/* Map the user-provided address to a sockaddr_un. */
+		struct sockaddr_un *sockaddr_un = (struct sockaddr_un *)sockaddr;
+
+		/* Starting at `sockaddr_un` we have the socket family and after it
+		 * the `sun_path`.
+		 *
+		 * Please note exceptions in the `sun_path`:
+		 * Taken from: https://man7.org/linux/man-pages/man7/unix.7.html
+		 *
+		 * An `abstract socket address` is distinguished (from a
+		 * pathname socket) by the fact that sun_path[0] is a null byte
+		 * ('\0').
+		 */
+
+		/* Check the exact point in which we have to start reading our path. */
+		unsigned long start_reading_point;
+		/* We skip the two bytes of socket family. */
+		char first_path_byte = *(char *)sockaddr_un->sun_path;
+		if(first_path_byte == '\0')
+		{
+			/* This is an abstract socket address, we need to skip the initial `\0`. */
+			start_reading_point = (unsigned long)sockaddr_un->sun_path + 1;
+		}
+		else
+		{
+			start_reading_point = (unsigned long)sockaddr_un->sun_path;
+		}
+
+		/* Pack the sockaddr info:
+		 * - socket family.
+		 * - socket_unix_path (sun_path).
+		 */
+		push__u8(auxmap->data, &auxmap->payload_pos, socket_family_to_scap(socket_family));
+		u16 written_bytes = push__charbuf(auxmap->data, &auxmap->payload_pos, start_reading_point, MAX_UNIX_SOCKET_PATH, KERNEL);
+		final_param_len = FAMILY_SIZE + written_bytes;
+		break;
+	}
+
+	default:
+		final_param_len = 0;
+		break;
+	}
+	push__param_len(auxmap->data, &auxmap->lengths_pos, final_param_len);
+}
+
+/**
+ * @brief Store socktuple info taken from kernel socket.
+ * We prefer extracting data directly from the kernel to
+ * obtain more precise information.
+ *
+ * Please note:
+ * In outbound connections `local` is the src while `remote` is the dest.
+ * In inbound connections vice versa.
+ *
+ * @param auxmap pointer to the auxmap in which we are storing the param.
+ * @param socket_fd socket from which we extract information about the tuple.
+ * @param direction specifies the connection direction.
+ */
+static __always_inline void auxmap__store_socktuple_param(struct auxiliary_map *auxmap, u32 socket_fd, int direction)
+{
+	u16 final_param_len = 0;
+
+	/* Get the socket family directly from the socket */
+	u16 socket_family = 0;
+	struct file *file = NULL;
+	file = extract__file_struct_from_fd(socket_fd);
```

Comment:
cosmetic: you could directly use `struct file *file = extract__....`

Reply:
ty